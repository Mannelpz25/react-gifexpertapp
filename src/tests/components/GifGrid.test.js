import React from 'react';
import { GifGrid } from "../../components/GifGrid"
import {shallow} from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas al componente GifGrid', () => {
    const category = "Hola Mundo"
    
    test('Debe mostrar el componente', () => {
       
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category ={category}/>);
        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar item cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category ={category}/>);        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })
    
    

})
