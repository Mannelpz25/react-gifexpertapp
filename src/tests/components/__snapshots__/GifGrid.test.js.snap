// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGrid Debe de mostrar item cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente GifGrid Debe mostrar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
